graylog2-web-interface/src/views/components/searchbar/RefreshControls.jsx

```diff
@@ -67,27 +89,36 @@ class RefreshControls extends React.Component<Props, State> {
     RefreshActions.setInterval(interval);
   };
 
+  _buttonLabel = (refreshConfigEnabled, naturalInterval) => {
+    let buttonText = 'Not updating';
+    if (refreshConfigEnabled) {
+      buttonText = <React.Fragment>Update every {naturalInterval}</React.Fragment>;
+    }
+    return <ButtonLabel>{buttonText}</ButtonLabel>;
+  }
+
   render() {
+    const { refreshConfig } = this.props;
     const intervalOptions = RefreshControls.INTERVAL_OPTIONS.map(([label, interval]: [string, number]) => {
       return <MenuItem key={`RefreshControls-${label}`} onClick={() => this._onChange(interval)}>{label}</MenuItem>;
     });
-    const intervalDuration = moment.duration(this.props.refreshConfig.interval);
+    const intervalDuration = moment.duration(refreshConfig.interval);
     const naturalInterval = intervalDuration.asSeconds() < 60
       ? <span>{intervalDuration.asSeconds()} <Pluralize singular="second" plural="seconds" value={intervalDuration.asSeconds()} /></span>
       : <span>{intervalDuration.asMinutes()} <Pluralize singular="minute" plural="minutes" value={intervalDuration.asMinutes()} /></span>;
-    const buttonLabel = <span>Update every {naturalInterval}</span>;
+    const buttonLabel = this._buttonLabel(refreshConfig.enabled, naturalInterval);
```

Comment:
I am not that happy with the naming of the var in combination with the function, but couldn't get up with a better one, what do you think?

Reply:
No sure what you are not happy about. But that's what functions are for :smile: :+1: